dask_glm/families.py

```diff
@@ -59,3 +60,40 @@ def gradient(Xbeta, X, y):
     @staticmethod
     def hessian(Xbeta, X):
         return 2 * dot(X.T, X)
+
+
+class Poisson(object):
+    """
+    This implements Poisson regression for count data. 
+    See https://en.wikipedia.org/wiki/Poisson_regression.
+    """
+
+    @staticmethod
+    def loglike(Xbeta, y):
+        eXbeta = exp(Xbeta)
+        yXbeta = y*Xbeta
+        return (eXbeta - yXbeta ).sum()
+
+    @staticmethod
+    def pointwise_loss(beta, X, y):
+        beta, y = beta.ravel(), y.ravel()
+        Xbeta = X.dot(beta)
+        return Poisson.loglike(Xbeta, y)
+
+    @staticmethod
+    def pointwise_gradient(beta, X, y):
+        beta, y = beta.ravel(), y.ravel()
+        Xbeta = X.dot(beta)
+        return Poisson.gradient(Xbeta, X, y)
+
+    @staticmethod
+    def gradient(Xbeta, X, y):
+        eXbeta = exp(Xbeta)
+        return dot(X.T, eXbeta - y)
+
+    @staticmethod
+    def hessian(Xbeta, X):
+        eXbeta = exp(Xbeta)
+        diag_eXbeta = da.diag(eXbeta)
+        x_diag_eXbeta = dot(diag_eXbeta, X)
```

Comment:
FWIW I think `eXbeta[:, None] * X` should broadcast appropriately and save some memory (avoids storing all the `0`'s from the `diag` creation).